src/QsCompiler/Transformations/SyntaxTreeTrimming.cs

```diff
@@ -11,8 +11,19 @@
 
 namespace Microsoft.Quantum.QsCompiler.Transformations.SyntaxTreeTrimming
 {
+    /// <summary>
+    /// Removes unused callables from the syntax tree.
+    /// </summary>
     public static class TrimSyntaxTree
     {
+        /// <summary>
```

Comment:
It might be worth calling out what happens to types and type constructors. (I think the types may be preserved but the type constructors are subject to the same trimming rules as other callables?)

Reply:
I was looking for that. I thought we had updated the code to prevent type constructors from being removed at some point. Do you remember?